src/util/OAuth2Util.js

```diff
@@ -65,6 +65,7 @@ util.getTokens = function (settings, params, controller) {
     }
 
     Util.triggerAfterError(controller, new Errors.OAuthError(error.message), settings);
+    settings.callGlobalError(error);
```

Comment:
triggerAfterError doesn't work because we are passing `error.message` instead of error ?

Reply:
- seems @sflanker concern didn't get error instead of actual content.?

Reply:
> Curious why cannot rely on `afterError` event?

Because that's not how Promises are supposed to work. If your API returns a promise expect that promise to eventually either resolve or reject. I shouldn't have to wire up some special event handler instead! Having some special event handler *in addition to* rejecting the promise would be fine if people need a special mechanism to subscribe to certain kinds of failure events.

Reply:
the default behavior of `settings.callGlobalError(error);`  is to throw error but user can customized it to do something different. 
shall we just throw errors?

Reply:
You're in a promise worker function so throwing exceptions would be sub optimal since that would also be un-handleable and leave the promise in a perpetually pending state. However that's not the behavior of `callGlobalError` in this context. It's behavior is to reject the promise, which it what it should be doing in this case.

I'm not an expert in your codebase, so it is possible that this same code path is invoked in other contexts where this behavior might not be ideal. This PR is just meant to be a suggestion.

Here's a simple repro of the issue: https://replit.com/@paulinvio/static-site-okta-signin#script.js
You get test the issue here: https://static-site-okta-signin.paulinvio.repl.co/
If you log in with valid credentials (I can provide you with some if you email me: paul@invioinc.com ), The result is no error or warning in the console, and the promise returned is forever stuck in a pending state (the catch handler is never called).